File: scratch/test_escaped_loc.py
import requests
import re

headers = {"User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64)"}

def test_user(u):
    r = requests.get(f"https://whop.com/@{u}", headers=headers)
    html = r.text
    
    # Match escaped or unescaped country / city
    country_m = re.search(r'\\?"country\\?"\s*:\s*\\?"([A-Za-z]{2})\\?"', html)
    city_m = re.search(r'\\?"city\\?"\s*:\s*\\?"([^\\"]+)\\?"', html)
    earned_m = re.search(r'(\$[\d,]+(?:\.\d+)?)\s*(?:<!--\s*-->\s*)*Earned', html)
    
    country = country_m.group(1) if country_m else None
    city = city_m.group(1) if city_m else None
    earned = earned_m.group(1) if earned_m else None
    
    print(f"@{u:15} -> Country: {country or 'Not Set':5} | City: {city or 'Not Set':20} | Earned: {earned or '—'}")

test_list = ["bonnielau", "dariuslewis32", "cameron", "jack", "moxyalili", "townhall", "scalewdreww", "therealr1cky"]
for u in test_list:
    test_user(u)

File: scratch/test_escaped_loc.ts
async function testEscapedProfileLocation(username: string) {
  const res = await fetch(`https://whop.com/@${username}`, {
    headers: {
      "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36",
    },
  });
  if (!res.ok) return null;
  const html = await res.text();

  // Match escaped or unescaped country and city near user object
  const countryMatch = html.match(/\\?"country\\?"\s*:\s*\\?"([A-Z]{2})\\?"/);
  const cityMatch = html.match(/\\?"city\\?"\s*:\s*\\?"([^\\"]+)\\?"/);

  const country = countryMatch ? countryMatch[1] : null;
  const city = cityMatch && cityMatch[1] !== "null" ? cityMatch[1] : null;

  return { username, country, city };
}

async function main() {
  const users = ["dwave6f", "townhall", "gtron", "iamrohit7", "founder-cca", "matteoriva", "shnayderman", "harviee"];
  for (const u of users) {
    const res = await testEscapedProfileLocation(u);
    console.log(res);
  }
}

main().catch(console.error);
